FR #1090: Add multi dataset support to FOCUS spec layout (#1337)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/glossary.md

> <a name="glossary:FOCUS-dataset"><b>FOCUS Dataset</b></a>

Reviewer: I think the definition of `FOCUS Dataset` needs to continue being _any_ of the datasets within FOCUS.  From there, we do need two other terms to delineate between the two types.

I would propose `fact` and `dimension`, but that doesn't align with our existing FOCUS terminology.

Thus, perhaps we use `transaction` and `reference` to refer to tables like `Cost and Usage` and `Contract`, respectively.  Or terms along those lines.

Author: I am open to suggestions here on what glossary terms we should include in the PR and the definitions they should have.

Reviewer 2: I would not use the terms fact and dimension terms as they aren't exactly a total fit. I would refer to it as FOCUS includes datasets:
- cost and usage dataset
- invoice dataset
- contract dataset

Reviewer: I think there's benefit down the road of having people understand we have a "main" transactional dataset, and some "supporting" reference tables.  But for simplicity and for now, skipping that is fine with.  So that means we would just strike the "FOCUS adjacent" section.

Reviewer 3: I would prefer avoiding "primary" vs. "adjacent" datasets. They're all FOCUS datasets. We just need to clarify which one. Some people may only care about an adjacent one for specific efforts, so I don't think those labels add any value.

Reviewer: Issued separate commit to address this.  Resolving.

In addition to these standardized columns, data generators may include custom provider-specific columns (prefixed with `x_`) where additional context is needed beyond what is captured in the defined FOCUS columns. If such custom columns introduce record-splitting (i.e., a single original charge results in multiple rows), the data generator is responsible for ensuring that all cost and quantity metrics still meet the aggregation and consistency rules required by the specification.

<a name="glossary:FOCUS-adjactent-dataset"><b>FOCUS Adjacent Dataset</b></a>

A FOCUS Adjacent Dataset is any dataset defined within the FOCUS specification that supports or augments the primary FOCUS dataset, but does not itself contain normalized cost and usage records.

These datasets are designed to provide additional context, metadata, mapping, or enrichment information that enhances the interpretability or completeness of the FOCUS dataset. Examples include datasets for pricing reference data, commitment discount mapping, or region definitions.

> FOCUS Adjacent Datasets follow the same principles of structure and consistency defined by the specification, including column naming conventions and formatting expectations.

Reviewer: Need to strike usage of "adjacent".  Just have an entry for FOCUS dataset.

> The FOCUS dataset is the primary dataset defined in the specification for capturing cost and usage information. It is one of several dataset types outlined in the FOCUS specification but serves as the central dataset for reporting normalized cost and usage records.
> 
> A FOCUS dataset is a structured collection of records that conforms to the BCP14 criteria established by FOCUS. All columns included must be defined in the FOCUS Columns section of the specification.
> 
> In addition to these standardized columns, data generators may include custom provider-specific columns (prefixed with `x_`) where additional context is needed beyond what is captured in the defined FOCUS columns. If such custom columns introduce record-splitting (i.e., a single original charge results in multiple rows), the data generator is responsible for ensuring that all cost and quantity metrics still meet the aggregation and consistency rules required by the specification.
> 
> <a name="glossary:FOCUS-adjactent-dataset"><b>FOCUS Adjacent Dataset</b></a>
> 
> A FOCUS Adjacent Dataset is any dataset defined within the FOCUS specification that supports or augments the primary FOCUS dataset, but does not itself contain normalized cost and usage records.
> 
> These datasets are designed to provide additional context, metadata, mapping, or enrichment information that enhances the interpretability or completeness of the FOCUS dataset. Examples include datasets for pricing reference data, commitment discount mapping, or region definitions.
> 
> FOCUS Adjacent Datasets follow the same principles of structure and consistency defined by the specification, including column naming conventions and formatting expectations.

Author: ```suggestion

A FOCUS dataset is a structured collection of records that conforms to the BCP14 criteria established by FOCUS. All columns included must be defined in the FOCUS Columns section of the specification.

In addition to these standardized columns, data generators may include custom provider-specific columns (prefixed with `x_`) where additional context is needed beyond what is captured in the defined FOCUS columns. If such custom columns introduce record-splitting (i.e., a single original charge results in multiple rows), the data generator is responsible for ensuring that all cost and quantity metrics still meet the aggregation and consistency rules required by the specification.

The collection of datasets are designed to provide billing insight, additional context, metadata, mapping, or enrichment information that enhances the interpretability or completeness.
```
